buildSrc/src/main/kotlin/source-sets-conventions.gradle.kts

```diff
@@ -30,13 +33,13 @@ kotlin {
 
     jvm {
         withJava()
-        compilations.configureEach {
-            kotlinOptions {
-                jvmTarget = "1.8"
-                freeCompilerArgs += "-Xjdk-release=1.8"
-            }
+        @OptIn(ExperimentalKotlinGradlePluginApi::class)
+        compilerOptions {
+            jvmTarget.set(JvmTarget.JVM_1_8)
+            freeCompilerArgs.add("-Xjdk-release=1.8")
         }
     }
+    jvmToolchain(jdkToolchainVersion)
```

Comment:
In master (2.0.20), KGP requires toolchain and and jvmTarget to be consistent. Otherwise, things like `org.gradle.api.InvalidUserCodeException: Inconsistent JVM-target compatibility detected for tasks 'compileTestJava' (1.8) and 'compileTestKotlin' (11).` pop up

Reply:
Hm, given that we have both `options.release = 8` in `JavaCompile` and `freeCompilerArgs.add("-Xjdk-release=1.8")` in `KotlinCompile`, do we actually need toolchains?

Reply:
Since you've added `jdkToolchainVersion` as a property, this block on line 21:
```
java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(11))
    }
}
```

probably should use it as well instead of `JavaLanguageVersion.of(11)`